hoodie-client/src/main/java/com/uber/hoodie/io/compact/HoodieCompactor.java

```diff
@@ -34,17 +32,31 @@
 
   /**
    * Compact the delta files with the data files
+   *
+   * @deprecated : Will be removed in next PR
    */
+  @Deprecated
   JavaRDD<WriteStatus> compact(JavaSparkContext jsc, final HoodieWriteConfig config,
       HoodieTable hoodieTable, String compactionCommitTime) throws Exception;
 
+  /**
+   * Generate workload for scheduling compaction
+   *
+   * @param jsc                  Spark Context
+   * @param hoodieTable          Hoodie Table
+   * @param config               Hoodie Write Configuration
+   * @param compactionCommitTime scheduled compaction commit time
+   * @return Compaction Workload
+   * @throws IOException when encountering errors
+   */
+  HoodieCompactionWorkload generateCompactionWorkload(JavaSparkContext jsc,
```

Comment:
this can be a private method ?

Reply:
This will be a top level API for scheduling compactions in the subsequent PR

Reply:
Yes, Clients need to schedule/run compactions through HoodieClient